### DISCUSSION TITLE
Search Box - clear "advanced"

### DISCUSSION THREAD

**Discussion by salascz** [microsoft-search/pnp-modern-search#2284] [Ideas]
Hello,
This is one of the behaviour our users are complaining about when using the Search Box

**Is your feature request related to a problem? Please describe.**

The use case:
1) Type query to Search Box
2) "Do anything" with results
3) "Go back" -> Clear the text in the search box
     Unfortunately, our users don't notice the clear ("X") button often - they use the backspace/delete on the keyboard and remove the sentence from input element. -> as a result, the clear ("X") button disappear but results are not refreshed.

> Note: Yes, you can type new query and click the clear button (it will appears again), but its and advanced for most of the users...

> Current solution does not seem to be intuitive enough :(

**Describe the solution you'd like**
I am aware, the visibility of the "Clear" button is driven by fluentui control and the button is not visible without value. 
Would it be possible to execute search query request (with empty query) when the previously used query was removed by user?

Thanks

**Comment by salascz** [comment]
Hello,
going back to this idea...

The SearchBox WebPart seems to be well prepared - previous user's input is included in the state `searchInputValue` + the search implementation block duplicate search calls in `this.props.onSearch(queryText);`

That said, the behaviour might be updated quite easily...

What is your opinion on following [code change](https://github.com/microsoft-search/pnp-modern-search/compare/main...salascz:pnp-modern-search:feature/update-empty-searchbox-behavior?expand=1)?  

Thanks

**Reply by wobba** [reply]
I think this looks ok actually. The 'side effect' would be that once a user manually clears the search box, a query will be executed. 

If you add a web part setting with **Re-query on clear**  that would be better. This way people who want the existing behavior which is the same as OOB search are not effected, and those who want to re-query can toggle it. What do you think?

**Comment by salascz** [comment]
Hi @wobba,
thanks for your feedback. It seems to be a good idea.

I tried to modify the code in my fork. (I am not native speaker of any supported language)

![image](https://user-images.githubusercontent.com/5662939/190091219-4c486ef1-f13e-4302-a512-f7c7a48840e9.png)
(disabled by default to keep current behaviour as default) 

Are there other parts I should follow before PR?
Thanks

**Reply by wobba** [reply]
Create a PR and we will review the code. Ensure you translate all labels to the supported languages we have.

**Reply by salascz** [reply]
Thanks, 

https://github.com/microsoft-search/pnp-modern-search/pull/2369
